### ISSUE TITLE
InvalidCastException when importing or exporting Paintings

### ISSUE THREAD

**Issue by ghost** [TexTools/FFXIV_TexTools_UI#34]
In version 2.3.4.19, attempting to export or import any Painting's textures results in the following exception followed by a full program crash.  

```
System.InvalidCastException: Unable to cast object of type 'xivModdingFramework.Items.DataContainers.XivFurniture' to type 'xivModdingFramework.Items.DataContainers.XivCharacter'.
   at FFXIV_TexTools.ViewModels.TextureViewModel.<Export>d__105.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at FFXIV_TexTools.Views.TextureView.<ExportTextureButton_Click>d__2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at FFXIV_TexTools.App.Main()
   ```
   
Similarly, attempting to use any of the Material buttons results in "An error occurred when trying to modify the material.  Error:Object reference not set to an instance of an object.", though the program recovers in these cases.

**Comment by Fhurai** [comment]
To add to the issue I encounter too, thoses messages
![image](https://user-images.githubusercontent.com/10893779/111287482-8d827500-8643-11eb-9513-79d443396ce4.png)
![image](https://user-images.githubusercontent.com/10893779/111287531-9b37fa80-8643-11eb-8ed9-1d38dcdc3cf5.png)
![image](https://user-images.githubusercontent.com/10893779/111287596-aa1ead00-8643-11eb-99f9-8621b8d34f28.png)

I try to import file to change the texture of the Voor Siran Painting by the way, if this can help

**Comment by Squall-Leonhart** [comment]
Still present in 2.3.4.23

**Comment by Luna664** [comment]
It shows me this message when I import since the last update

![image](https://user-images.githubusercontent.com/75101727/115973641-e30d4400-a556-11eb-9f8c-2c83bd888578.png)

**Comment by Fhurai** [comment]
Confirmation of the problem pointed above.
The importation menu can be opened for a painting but when the file to import is selected and the button to import is clicked, cast error is shown.

**Comment by Lunaretic** [comment]
Previous patch seems to have fixed this for export, but not Import.

**Comment by Lunaretic** [comment]
Fixed in 2.3.5.9
